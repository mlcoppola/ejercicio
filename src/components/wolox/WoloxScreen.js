import React, { useContext, useEffect, useState } from 'react';
import { useHistory } from 'react-router-dom';
import { AuthContext } from '../../auth/AuthContext';
import { useFetch } from '../../hooks/useFetch';
import { types } from '../../types/types';
import { Listado } from './Listado';

export default function WoloxScreen() {

    const url = 'http://private-8e8921-woloxfrontendinverview.apiary-mock.com/techs';
    const { sortedData } = useFetch(url);
    const [techs, setTechs] = useState([]);
    const [inputValue, setInputValue] = useState('');
    const { user: { name }, dispatch } = useContext(AuthContext);
    const history = useHistory();

    const handleLogout = () => {

        history.replace('/auth/login');

        dispatch({
            type: types.logout
        });
    }

    const handleInputChange = (e) => {
        let value = e.target.value.toLowerCase();
        setInputValue(value);
    }

    const filterTechs = () => (
        techs.filter(tech =>
            tech.tech.toLowerCase().includes(inputValue)
            || tech.type.toLowerCase().includes(inputValue)
        )
    )

    useEffect(() => {
        if (sortedData) {
            setTechs(sortedData);
        }
    }, [sortedData])

    const setOrder = (e) => {
        let orden = e.target.value;

        let sorted = techs.sort((a, b) => {
            const isReversed = (orden === 'asc' || orden === '') ? 1 : -1;
            return isReversed * a.tech.localeCompare(b.tech);
        });

        setTechs([...sorted]);
    }

    return (
        <>

            <div className="listado__contenedor-superior">
                <span className="listado__bienvenida">Bienvenido {name}</span>


                <input
                    type="text"
                    placeholder="Buscar tecnología"
                    value={inputValue}
                    onChange={handleInputChange}
                    className="listado__buscador"
                />

                <div className="listado__orden">
                    <div className="listado__row">
                        <input type="radio" label="asc" name="order" value="asc" defaultChecked={true} onClick={setOrder} /> Ascendente
                    </div>
                    <div className="listado__row">
                        <input type="radio" label="asc" name="order" value="desc" onClick={setOrder} /> Descendente
                    </div>
                </div>


                <button
                    onClick={handleLogout}
                    className="listado__boton-logout"
                >
                    Logout
                </button>

            </div>

            <div className="listado__contenedor-tecnologias animate__animated animate__fadeIn">

                {
                    filterTechs().map(tech => (
                        <Listado
                            key={tech.tech}
                            tech={tech}
                        />
                    ))
                }

                <div className="listado__float">Cantidad de tecnologías: {filterTechs().length}</div>
            </div>
        </>
    )
}
