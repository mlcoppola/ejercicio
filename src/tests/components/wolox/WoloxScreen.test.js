/* eslint-disable no-undef */
import React from 'react';

import { shallow } from 'enzyme';
import WoloxScreen from '../../../components/wolox/WoloxScreen';

jest.mock('react', () => {
    const ActualReact = require.requireActual('react');
    return {
        ...ActualReact,
        useContext: () => ({ user: {} }),
    };
});

describe('Pruebas en <WoloxScreen />', () => {
    test('debe de mostrar el componente correctamente', () => {
        const wrapper = shallow(<WoloxScreen />);

        expect(wrapper).toMatchSnapshot();
    });

})
