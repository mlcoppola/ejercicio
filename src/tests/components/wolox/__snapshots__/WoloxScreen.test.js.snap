// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <WoloxScreen /> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <div
    className="listado__contenedor-superior"
  >
    <span
      className="listado__bienvenida"
    >
      Bienvenido 
    </span>
    <input
      className="listado__buscador"
      onChange={[Function]}
      placeholder="Buscar tecnología"
      type="text"
      value=""
    />
    <div
      className="listado__orden"
    >
      <div
        className="listado__row"
      >
        <input
          defaultChecked={true}
          label="asc"
          name="order"
          onClick={[Function]}
          type="radio"
          value="asc"
        />
         Ascendente
      </div>
      <div
        className="listado__row"
      >
        <input
          label="asc"
          name="order"
          onClick={[Function]}
          type="radio"
          value="desc"
        />
         Descendente
      </div>
    </div>
    <button
      className="listado__boton-logout"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
  <div
    className="listado__contenedor-tecnologias"
  >
    <div
      className="listado__float"
    >
      Cantidad de tecnologías: 
      0
    </div>
  </div>
</Fragment>
`;
